PickerDesktopHosting/XAMLHostingDesktop/XAMLHostingDesktop.cpp

```diff
@@ -0,0 +1,225 @@
+
+    return RegisterClassExW(&wcex);
+}
+
+//
+//   FUNCTION: InitInstance(HINSTANCE, int)
+//
+//   PURPOSE: Saves instance handle and creates main window
+//
+//   COMMENTS:
+//
+//        In this function, we save the instance handle in a global variable and
+//        create and display the main program window.
+//
+BOOL InitInstance(HINSTANCE hInstance, int nCmdShow)
+{
+    hInst = hInstance; // Store instance handle in our global variable
+
+    HWND hWnd = CreateWindowW(szWindowClass, szTitle, WS_OVERLAPPEDWINDOW,
+        CW_USEDEFAULT, 0, CW_USEDEFAULT, 0, nullptr, nullptr, hInstance, nullptr);
+
+    if (!hWnd)
+    {
+        return FALSE;
+    }
+
+    // Begin XAML Islands walkthrough code.
+    if (_desktopWindowXamlSource != nullptr)
+    {
+        auto interop = _desktopWindowXamlSource.as<IDesktopWindowXamlSourceNative>();
+        check_hresult(interop->AttachToWindow(hWnd));
+        HWND hWndXamlIsland = nullptr;
+        interop->get_WindowHandle(&hWndXamlIsland);
+        RECT windowRect;
+        ::GetWindowRect(hWnd, &windowRect);
+        ::SetWindowPos(hWndXamlIsland, NULL, 0, 0, windowRect.right - windowRect.left, windowRect.bottom - windowRect.top, SWP_SHOWWINDOW);
+        _myUserControl = winrt::Control::FilePickerControl();
+        _desktopWindowXamlSource.Content(_myUserControl);
+    }
+    // End XAML Islands walkthrough code.
+
+    ShowWindow(hWnd, nCmdShow);
+    UpdateWindow(hWnd);
+
+    return TRUE;
+}
+
+void AdjustLayout(HWND hWnd)
+{
+    if (_desktopWindowXamlSource != nullptr)
+    {
+        auto interop = _desktopWindowXamlSource.as<IDesktopWindowXamlSourceNative>();
+        HWND xamlHostHwnd = NULL;
+        check_hresult(interop->get_WindowHandle(&xamlHostHwnd));
+        RECT windowRect;
+        ::GetWindowRect(hWnd, &windowRect);
+        ::SetWindowPos(xamlHostHwnd, NULL, 0, 0, windowRect.right - windowRect.left, windowRect.bottom - windowRect.top, SWP_SHOWWINDOW);
+    }
+}
+
+//
+//  FUNCTION: WndProc(HWND, UINT, WPARAM, LPARAM)
+//
+//  PURPOSE: Processes messages for the main window.
+//
+//  WM_COMMAND  - process the application menu
+//  WM_PAINT    - Paint the main window
+//  WM_DESTROY  - post a quit message and return
+//
+//
+LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    switch (message)
+    {
+    case WM_COMMAND:
+        {
+            int wmId = LOWORD(wParam);
+            // Parse the menu selections:
+            switch (wmId)
+            {
+            case IDM_ABOUT:
+                DialogBox(hInst, MAKEINTRESOURCE(IDD_ABOUTBOX), hWnd, About);
+                break;
+            case IDM_EXIT:
+                DestroyWindow(hWnd);
+                break;
+            default:
+                return DefWindowProc(hWnd, message, wParam, lParam);
+            }
+        }
+        break;
+    case WM_PAINT:
+        {
+            PAINTSTRUCT ps;
+            HDC hdc = BeginPaint(hWnd, &ps);
```

Comment:
One of wil::unique_hdc*? https://github.com/microsoft/wil/wiki/RAII-resource-wrappers#available-unique_any-handle-types